plugins/pulp_rpm/plugins/distributors/yum/metadata/updateinfo.py

```diff
@@ -133,13 +133,16 @@ def _get_package_checksum_tuple(self, package):
 
         return package_checksum_tuple
 
-    def add_unit_metadata(self, item):
+    def add_unit_metadata(self, item, filtered_pkglist):
```

Comment:
I wonder if `filtered_pkglist` should default to None here, and if it's None, be set to `item.pkglist` in the method body. This would reduce the amount of changes introduced by this change, particularly in the tests where most of the calls to this method look like `add_unit_metadata(errata, errata.pkglist)`. There is already coverage for both branches, where filtered_pkglist is just the unaltered pkglist, as well as when it's a filterered pkglist, so this shouldn't expose a need for additional tests.